src/app/myworkflows/sidebar-accordion/github-apps-logs/github-apps-logs.component.ts

```diff
@@ -66,20 +85,129 @@ export class GithubAppsLogsComponent implements OnInit {
       : ['repository', 'entryName', 'deliveryId', 'reference', 'success', 'type'];
     this.displayedColumns = ['eventDate', 'githubUsername', ...this.columnsToDisplay];
     this.loading = true;
-    this.dataSource.sort = this.sort;
-    this.dataSource.paginator = this.paginator;
-    const lambdaEvents = this.matDialogData.userId
-      ? this.lambdaEventsService.getUserLambdaEvents(this.matDialogData.userId)
-      : this.lambdaEventsService.getLambdaEventsByOrganization(this.matDialogData.organization);
+  }
+
+  ngAfterViewInit() {
+    this.loadAppsLogs();
+  }
+
+  loadAppsLogs() {
+    setTimeout(() => {
+      this.dataSource.sort = this.sort;
+      this.pageSize$ = this.paginatorQuery.eventPageSize$;
+      this.pageIndex$ = this.paginatorQuery.eventPageIndex$;
+
+      // Initial load
+      this.loadEvent(this.matDialogData, this.paginator.pageIndex, this.paginator.pageSize, null, null, null);
+      this.paginatorService.setPaginator(this.type, this.paginator.pageSize, this.paginator.pageIndex);
+
+      // Handle paginator changes
+      merge(this.paginator.page)
+        .pipe(distinctUntilChanged())
+        .subscribe(() => {
+          this.loadEvent(
+            this.matDialogData,
+            this.paginator.pageIndex * this.paginator.pageSize,
```

Comment:
This code appears to use `pageIndex` in two different ways.  At line 101, we call `loadEvent` with the raw `pageIndex` value, and here, we multiply `pageIndex` by the `pageSize`.  It should probably be consistently one or the other, and if we used the db code as a guide, an "index" would be an offset from 0, measured in the number of results.

Reply:
Yeah, sorry, I think I misinterpreted the code, the `pageIndex` is coming from `MatPaginator`, which [I assume] defines the page index as the number of pages of items.  Would still recommend that in the first `loadEvents` call, if you use `pageIndex` rather than zero, you multiply it by `pageSize` as you do here, just in case it's not zero for some reason.